gui/src/WorkspaceView/controls/SurfaceSelectionControls.tsx

```diff
@@ -0,0 +1,164 @@
+import { Checkbox } from '@material-ui/core';
+import React, { FunctionComponent, useCallback, useMemo } from 'react';
+import { WorkspaceSurfaceScalarField, WorkspaceSurfaceVectorField, WorkspaceViewData } from 'VolumeViewData';
+import { WorkspaceViewSelection, WorkspaceViewSelectionAction } from '../workspaceViewSelectionReducer';
+
+const separator = '::'
+
+type SurfaceSelectionControlProps = {
+    data: WorkspaceViewData
+    selection: WorkspaceViewSelection
+    selectionDispatch: (a: WorkspaceViewSelectionAction) => void
+}
+
+type PerSurfaceControlsProps = {
+    name: string
+    displayed: boolean
+    scalarFields: WorkspaceSurfaceScalarField[]
+    vectorFields: WorkspaceSurfaceVectorField[]
+    selectedScalarFieldName?: string
+    selectedVectorFieldName?: string
+    visibilityCallback: any
+    selectedVectorFieldCallback: any
+    selectedScalarFieldCallback: any
+}
+
+type FieldSelectorProps = {
+    surfaceName: string
+    fields: WorkspaceSurfaceScalarField[] | WorkspaceSurfaceVectorField[]
+    selectedFieldName?: string
+    type: 'vector' | 'scalar'
+    callback: any
+}
+
+type ChangeEvent = React.ChangeEvent<{value: unknown}>
+
+const SurfaceSelectionControls: FunctionComponent<SurfaceSelectionControlProps> = (props: SurfaceSelectionControlProps) => {
+    const { data, selection, selectionDispatch } = props
+
+    const handleToggleVisibleSurface = useCallback((e: ChangeEvent) => {
+        const surfaceName = e.target.value as string
+        selectionDispatch({
+            type: 'toggleVisibleSurface',
+            surfaceName
+        })
+    }, [selectionDispatch])
+
+    const fieldTtoggler = useCallback((name: string, fieldType: 'scalar' | 'vector') => {
+        const type = fieldType === 'scalar' ? 'toggleSelectedSurfaceScalarField' : 'toggleSelectedSurfaceVectorField'
+        const [ surfaceName, surfaceFieldName ] = name.split(separator)
+        selectionDispatch({ type, surfaceName, surfaceFieldName })
+    }, [selectionDispatch])
```

Comment:
The same function can handle dispatching events for both vector and scalar fields (which is further simplified since I gave the field name the same key for both types).
There is probably a way to avoid having to create the two wrapper callback functions below, but the examples I saw used anonymous functions where the callback is registered, and I was concerned about recreating a lot of cruft on every render.

Below, when field checkboxes are rendered, the name of the surface and the field will be combined as the value of the checkbox. Line 49 splits them back out. (It'd be unfortunate if a field or surface name already contained our separator, but I'm not worrying about it right now, and there's possible fixes if it does pose a problem.)